verification/curator-service/api/src/clients/aws-batch-client.ts

```diff
@@ -0,0 +1,135 @@
+ * connection configuration (including mocking) should occur on the AWS object
+ * before construction.
+ */
+export default class AwsBatchClient {
+    private readonly batchClient: AWS.Batch;
+    constructor(
+        private readonly serviceEnv: string,
+        readonly jobQueueArn: string,
+        awsRegion: string,
+    ) {
+        AWS.config.update({ region: awsRegion });
+        this.batchClient = new AWS.Batch({
+            apiVersion: '2016-08-10',
+        });
+    }
+
+    /**
+     * Submit retrieval batch job synchronously, returning its output.
+     *
+     * @param sourceId - ID of the source to be retrieved.
+     * @param parseDateRange - Range over which to perform parsing (inclusive).
+     */
+    doRetrieval = async (
+        sourceId: string,
+        parseDateRange?: ParseDateRange,
+    ): Promise<RetrievalPayload> => {
+        try {
+        	const r = await this.batchClient
+                .describeJobDefinitions({ maxResults: 10000 })
+                .promise();
+
+            let jobName;
+            jobDefsLoop:
+	          	for (var jd of r.jobDefinitions || []) {
+                    if (jd.containerProperties){
+                        if (jd.containerProperties.environment) {
+                            for (var env_var of jd.containerProperties.environment) {
+                                if (env_var.name == 'EPID_INGESTION_SOURCE_ID' && env_var.value == sourceId) {
+                                    jobName = jd.jobDefinitionName;
+                                    break jobDefsLoop;
+                                }
+                            }
+                        }
+                    }
+	          	}
+
+	        if (!jobName) {
+	        	throw Error(
+	        		`Could not find jobDefinition for sourceId ${sourceId}`
+	        	);
+	        }
+
+            const res = await this.batchClient
+        		.submitJob({
+        			jobDefinition: jobName + '-' + this.serviceEnv,
+        			jobName: jobName,
+        			jobQueue: this.jobQueueArn,
+        			containerOverrides: parseDateRange
+            			? {
+            				environment: [
+            					{
+            						name: 'EPID_INGESTION_PARSING_DATE_RANGE',
+            						value: parseDateRange.start.concat(',', parseDateRange.end)
+            					}
+            				]
+            			  }
+            			: undefined
+        		})
+                .promise();
+            let response = res as any;
+            if (response.$metadata.HTTPStatusCode != 200){
+            	logger.error(response);
+            	throw Error(
+                    `Retrieving source "${sourceId}" content: ${response.$metadata}`,
+                );
+            }
+            let ret = {jobName: res.jobName};
+            return ret;
+        } catch (e) {
+            logger.error(e);
+            throw e;
+        }
+    };
+
+    /** Lists the available parsers */
+    listParsers = async (): Promise<BatchJobDefinition[]> => {
```

Comment:
Yeah, this gets used by `ui/src/components/ParsersAutocomplete.tsx`.
When it lived in the lambda client it filtered on functions with `FunctionName?.includes('ParsingFunction')`.
Rather than filter on "ingestor", would it be better to filter on "${env}"?